crates/multi-transaction-payment/src/mock.rs

```diff
@@ -0,0 +1,266 @@
+        DexGeneral: dex_general::{Pallet, Call, Storage, Event<T>},
+        Currency: currency::{Pallet},
+        Tokens: orml_tokens::{Pallet, Storage, Event<T>, Config<T>},
+        MultiTransactionPayment: multi_transaction_payment::{Call, Pallet, Storage},
+        TransactionPayment: pallet_transaction_payment::{Pallet, Storage, Event<T>},
+        Testing: testing_helpers::{Call},
+    }
+);
+
+#[frame_support::pallet]
+pub mod testing_helpers {
+    use frame_support::{
+        dispatch::{DispatchErrorWithPostInfo, PostDispatchInfo},
+        pallet_prelude::*,
+    };
+    use frame_system::pallet_prelude::*;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {}
+
+    #[pallet::error]
+    pub enum Error<T> {
+        Fail,
+    }
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight({
+            *_expected
+		})]
+        #[frame_support::transactional]
+        pub fn weighted(
+            _origin: OriginFor<T>,
+            _expected: Weight,
+            actual_weight: Option<Weight>,
+            err: bool,
+            pays_fee: Pays,
+        ) -> DispatchResultWithPostInfo {
+            let post_info = PostDispatchInfo {
+                actual_weight,
+                pays_fee,
+            };
+            if err {
+                Err(DispatchErrorWithPostInfo {
+                    error: Error::<T>::Fail.into(),
+                    post_info,
+                })
+            } else {
+                Ok(post_info)
+            }
+        }
+    }
+}
+
+pub struct DealWithFees<T, GetCurrencyId, GetAccountId>(PhantomData<(T, GetCurrencyId, GetAccountId)>);
+
+impl<T, GetCurrencyId, GetAccountId> OnUnbalanced<NegativeImbalance<T, GetCurrencyId>>
+    for DealWithFees<T, GetCurrencyId, GetAccountId>
+where
+    T: orml_tokens::Config,
+    GetCurrencyId: Get<T::CurrencyId>,
+    GetAccountId: Get<T::AccountId>,
+{
+    fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalance<T, GetCurrencyId>>) {
+        if let Some(mut fees) = fees_then_tips.next() {
+            if let Some(tips) = fees_then_tips.next() {
+                tips.merge_into(&mut fees);
+            }
+            orml_tokens::CurrencyAdapter::<T, GetCurrencyId>::resolve_creating(&GetAccountId::get(), fees);
+        }
+    }
+}
+
+pub struct PairLpIdentity;
+impl GenerateLpAssetId<CurrencyId> for PairLpIdentity {
+    fn generate_lp_asset_id(asset_0: CurrencyId, asset_1: CurrencyId) -> Option<CurrencyId> {
+        CurrencyId::join_lp_token(asset_0, asset_1)
+    }
+}
+
+pub struct CurrencyConvert;
+impl currency::CurrencyConversion<currency::Amount<Test>, CurrencyId> for CurrencyConvert {
+    fn convert(
+        _amount: &currency::Amount<Test>,
+        _to: CurrencyId,
+    ) -> Result<currency::Amount<Test>, sp_runtime::DispatchError> {
+        unimplemented!()
```

Comment:
Paying with lend tokens would increase the max weight, but not by much

Reply:
That should be part of the dex swap weight